Core/Main/itkElastixRegistrationMethod.hxx

```diff
@@ -177,6 +177,25 @@ ElastixRegistrationMethod<TFixedImage, TMovingImage>::GenerateData()
                                  m_EnableOutput && m_LogToConsole,
                                  static_cast<elastix::log::level>(m_LogLevel));
 
+  if (m_InitialTransformParameterObject && !m_OutputDirectory.empty())
+  {
+    // Write InitialTransformParameters.0.txt, InitialTransformParameters.1.txt, InitialTransformParameters.2.txt, etc.
+    std::string initialTransformParameterFileName = "NoInitialTransform";
+    unsigned    i{};
+
+    for (auto transformParameterMap : m_InitialTransformParameterObject->GetParameterMaps())
+    {
+      transformParameterMap["InitialTransformParameterFileName"] = { initialTransformParameterFileName };
+
+      const auto transformParameterFileName = "InitialTransformParameters." + std::to_string(i) + ".txt";
+      elx::ParameterObject::WriteParameterFile(transformParameterMap, m_OutputDirectory + transformParameterFileName);
+      initialTransformParameterFileName = transformParameterFileName;
+      ++i;
+    }
+
+    argumentMap["-tp"] = initialTransformParameterFileName;
```

Comment:
It appears sufficient to place the name of the _last_ initial transform parameter file (for example, "InitialTransformParameters.2.txt") in the command-line argument map, entry "-tp". This is picked up by `TransformBase<TElastix>::ReadFromFile()` at https://github.com/SuperElastix/elastix/blob/007daddab591a88875cd4ba141936ac5a027838e/Core/ComponentBaseClasses/elxTransformBase.hxx#L444